src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -751,26 +754,71 @@ export const updateForm = (
  *
  * @returns ok(collaborators) if form updates successfully
  * @returns err(PossibleDatabaseError) if any database errors occurs
+ * @returns err(InvalidCollaboratorError) if a newly-added collaborator email is not whitelisted
  */
 export const updateFormCollaborators = (
   form: IPopulatedForm,
   updatedCollaborators: FormPermission[],
-): ResultAsync<FormPermission[], PossibleDatabaseError> => {
+): ResultAsync<
+  FormPermission[],
+  PossibleDatabaseError | InvalidCollaboratorError
+> => {
+  const logMeta = {
+    action: 'updateFormCollaborators',
+    formId: form._id,
+  }
+
+  // Get the updated (added or modified) collaborator emails (i.e. they are not
+  // in the original collaborator list).
+  const updatedCollaboratorEmails = updatedCollaborators
+    .filter(
+      (c1) =>
+        !form.permissionList.some(
+          (c2) => c1.email === c2.email && c1.write === c2.write,
```

Comment:
Just checking: is normalization of the emails (trim + lowercasing) done before this filtering occurs? (question is also relevant to the agency search below)

Reply:
these are both done in the frontend. we have validation middleware requiring that the emails be emails (which means they must already be trimmed), but i don't think this enforces a lowercase restriction. we should do it here too. Thanks for catching! 

I'll open a PR to force lowercase validation in the middleware